src/scanning/scan.ts

```diff
@@ -66,12 +65,11 @@ export async function scan(
   };
 
   // The callback passed to `scan()`
-  function visit(visitor: any) {
+  function visit(visitor: V) {
     visitors.push(visitor);
     if (!runner) {
       runner = Promise.resolve().then(runVisitors).catch((error) => {
         visitError = visitError || error;
-        throw error;
```

Comment:
This is the only change that matters much. The rest were just for my own benefit.